operators/pkg/custom-metrics-server/container.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020-2022 Politecnico di Torino
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package custom_metrics_server
+
+import (
+	"context"
+
+	// runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1".
+	"github.com/go-logr/logr"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+// listContainers calls ListConteiners which sends a ListContainerRequest to the server, and parses
+// the returned ListContainerResponse.
+func listContainers(ctx context.Context, log logr.Logger, id, podID *string, state *runtimeapi.ContainerState, labels map[string]string) ([]*runtimeapi.Container, error) {
```

Comment:
Delete all unused parameters, and pass the others by value, to keep the logic simple.
Once more, this wrapper seems practically useless to me, anc could be merged with runtimeClient.ListContainers.